### PR TITLE
DB-9388 write complete update row when updating the sys.sysstatements…

### PR DESCRIPTION
**Pull request by yxia92**
… table

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java [hunk 1]
```diff
@@ -3471,20 +3471,21 @@ public void clearSPSPlans() throws StandardException{
 
         FormatableBitSet columnToReadSet=new FormatableBitSet(SYSSTATEMENTSRowFactory.SYSSTATEMENTS_COLUMN_COUNT);
         FormatableBitSet columnToUpdateSet=new FormatableBitSet(SYSSTATEMENTSRowFactory.SYSSTATEMENTS_COLUMN_COUNT);
-        columnToUpdateSet.set(SYSSTATEMENTSRowFactory.SYSSTATEMENTS_VALID-1);
-        columnToUpdateSet.set(SYSSTATEMENTSRowFactory.SYSSTATEMENTS_CONSTANTSTATE-1);
-
+        for(int i=0;i<SYSSTATEMENTSRowFactory.SYSSTATEMENTS_COLUMN_COUNT;i++){
+            if (i+1==SYSSTATEMENTSRowFactory.SYSSTATEMENTS_CONSTANTSTATE) {
+                columnToUpdateSet.set(i);
```
Comment (martinrupp) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:3476]: why is `columnToUpdateSet.set(i);` done in if and in else?
shouldn't it rather be

```
       for(int i=0;i<SYSSTATEMENTSRowFactory.SYSSTATEMENTS_COLUMN_COUNT;i++){
            if (i+1 != SYSSTATEMENTSRowFactory.SYSSTATEMENTS_CONSTANTSTATE) {
                columnToReadSet.set(i);
            }
            columnToUpdateSet.set(i);
        }
```

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java [hunk 2]
```diff
@@ -3502,6 +3503,14 @@ public void clearSPSPlans() throws StandardException{
 
         while(sc.fetchNext(rowTemplate)){
             /* Replace the column in the table */
+            for (int i=0; i<rowTemplate.length; i++) {
+                if (i  == SYSSTATEMENTSRowFactory.SYSSTATEMENTS_VALID-1)
+                    replaceRow[i] = new SQLBoolean(false);
+                else if (i == SYSSTATEMENTSRowFactory.SYSSTATEMENTS_CONSTANTSTATE-1)
```
Comment (martinrupp) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:3509]: rest of file uses style `i+1 == SYSSTATEMENTSRowFactory.SYSSTATEMENTS_CONSTANTSTATE`, i think we should stick to one style (same line 3507)

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/SpliceCatalogUpgradeScripts.java [hunk 3]
```diff
@@ -96,7 +96,7 @@ public void run() throws StandardException{
         NavigableSet<Splice_DD_Version> keys=scripts.navigableKeySet();
         for(Splice_DD_Version version : keys){
             if(currentVersion!=null){
-                if(ddComparator.compare(version,currentVersion)<0){
+                if(ddComparator.compare(version,currentVersion)<=0){
```
Comment (martinrupp) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/SpliceCatalogUpgradeScripts.java:99]: i'm new, how is this related to the change in sys.sysstatements ?

Comment (yxia92) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/SpliceCatalogUpgradeScripts.java:99]: @martinrupp You are right, this change is not related to the update for the sys.sysstatements table. It is a fix for a minor issue with our upgrade process in general. 
Each upgrade script is associated with a version (let's call it S) which is usually the database version this script is introduced. 
In principal, if the currently installed database version (let's call it C) is already equal or above the version S, we don't need to call this upgrade script as the change should have already been applied. 
Before the fix, the upgrade script is still possible to be executed when C is the same as S.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java (hunk 2 above)
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java]: I made the change to make the style consistent. Thanks!

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java (hunk 1 above)
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java]: Done!
